### PR TITLE
tests: Cleanup for Job e2e tests global vars

### LINKED ISSUE #121689: Cleanup for Job e2e tests: do not use global vars for parallelism, completions and backoffLimit.
/kind cleanup 
Reasons:
1. the global vars require make the reading hard as require jumping in the long file
2. overriding the values by mistake in one test can cause another test to fail
3. the global values (parallelism=2, completions=4) could be smaller for some tests, but they are reused because this is less effort. For example here: "Using a pod failure policy to not count some failures towards the backoffLimit" completion could be just 2, making the run faster.

For 1 and 2 see discussion here: https://github.com/kubernetes/kubernetes/pull/121633#discussion_r1377588047

### PR DESCRIPTION
**Pull request by HuSharp**
#### Which issue(s) this PR fixes:

Fixes #121689

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/apps/job.go [hunk 1]
```diff
@@ -71,23 +71,16 @@ type watchEventConfig struct {
...
-	largeParallelism := int32(90)
-	largeCompletions := int32(90)
-
-	backoffLimit := int32(6) // default value
 
 	// Simplest case: N pods succeed
 	ginkgo.It("should run a job to completion when tasks succeed", func(ctx context.Context) {
 		ginkgo.By("Creating a job")
-		job := e2ejob.NewTestJob("succeed", "all-succeed", v1.RestartPolicyNever, parallelism, completions, nil, backoffLimit)
+		job := e2ejob.NewTestJob("succeed", "all-succeed", v1.RestartPolicyNever, int32(2), int32(4), nil, int32(6))
```
Comment (mimowo) [inline test/e2e/apps/job.go:78]: I would suggest to still keep the needed parameters named locally. Two reasons,
- when passed to the function it is not clear what they mean
- the completions =4 needs to match with the line below `err = e2ejob.WaitForJobComplete(ctx, f.ClientSet, f.Namespace.Name, job.Name, int32(4))`.

Comment (HuSharp) [reply test/e2e/apps/job.go]: fixed. PTAL :)

Comment (mimowo) [reply test/e2e/apps/job.go]: Thanks, but the build fails currently, please fix.

Comment (HuSharp) [reply test/e2e/apps/job.go]: ci passed, PTAL again, thx :)

File: go.sum [hunk 2]
```diff
@@ -28,143 +28,28 @@ cloud.google.com/go v0.90.0/go.mod h1:kRX0mNRHe0e2rC6oNakvwQqzyDmg57xJ+SZU1eT2aD
 cloud.google.com/go v0.93.3/go.mod h1:8utlLll2EF5XMAV15woO4lSbWQlk8rer9aLOfLh7+YI=
 cloud.google.com/go v0.94.1/go.mod h1:qAlAugsXlC+JWO+Bke5vCtc9ONxjQT3drlTTnAplMW4=
 cloud.google.com/go v0.97.0/go.mod h1:GF7l59pYBVlXQIBLx3a761cZ41F9bBH3JUlihCt2Udc=
-cloud.google.com/go v0.110.6/go.mod h1:+EYjdK8e5RME/VY/qLCAtuyALQ9q67dvuum8i+H5xsI=
```
Comment (mimowo) [inline go.sum:31]: Please revert

Comment (HuSharp) [reply go.sum]: done

**Review by mimowo — COMMENTED** [review]
/lgtm
/assign @soltysh

File: test/e2e/apps/job.go (hunk 1 above)
Comment (mimowo) [reply test/e2e/apps/job.go]: lgtm

**Review by soltysh — APPROVED** [review]
/approve

File: test/e2e/apps/job.go (hunk 1 above)
Comment (raminderis) [reply test/e2e/apps/job.go]: > ci passed, PTAL again, thx :) cc @soltysh

@HuSharp I wanted to understand what it means by "CI passed" in this case. did you run "go test" locally to ensure your changing were successful? - is that what you mean by "CI passed" or is there a CI pipeline that we can leverage for such verifications before code merge? Trying to educate myself. Thanks in advance.

**Pull request merged**